TechtonicaModLoader/Models/ProfileManager.cs

```diff
@@ -0,0 +1,95 @@
+﻿using CommunityToolkit.Mvvm.ComponentModel;
+using Newtonsoft.Json;
+using System.Collections.ObjectModel;
+using System.IO;
+using TechtonicaModLoader.Stores;
+
+namespace TechtonicaModLoader.Models
+{
+    public partial class ProfileManager : ObservableObject, IProfileManager
+    {
+        // Members
+        private readonly Dictionary<int, Profile> profiles = new();
+
+        // Properties
+
+        public Profile ActiveProfile {
+            get => profiles[Settings.UserSettings.ActiveProfileID.Value];
+            set {
+                if (value == null) return;
+                Settings.UserSettings.ActiveProfileID.Value = value.Id;
+                OnPropertyChanged(nameof(ActiveProfile));
+            }
+        }
+
+        public ObservableCollection<Profile> ProfilesList { get; } = new();
+
+        // Public Functions
+
+        public void CreateNewProfile(string name) {
+            if (ProfilesList.Select(profile => profile.Name).Contains(name)) {
+                throw new ArgumentException($"The profile '{name}' already exists.");
```

Comment:
Since this a part of the Model layer, this is the best course here.   The ViewModel can either protect from this ever happening or try/catch.  I prefer the former and have organized the code that way.

Reply:
Partially answered above. 

This is where you have to yield to the singular view of MVVM.  There are only three universes everything lives in.  Thinking that somehow a service is a way around this leads to colliding of universes and the summoning of the Old Gods who will point and shake their heads at the folly.  

OK, joking aside, MVVM is an architecture and a set of principles.  That's it.  It doesn't define implementation specifics and only mandates that there are three layers and the layers have a set interaction model.    The fact that something is implemented as a service or some other implementation is irrelevant:  it should always follow the architecture.    Anything that allows for interactions outside the definitions of MVVM means that the _entire_ code involved does not follow a MVVM architecture.   Full Stop.  No pleas for clemency. 

So if you are still with me, then we can look at ProfileManager through that lens and hopefully recognize that it holds application data and interacts with external things to read and write that data. So from the strict view of MVVM, it is part of the Model.  

Since you brought it up but previously had mentioned you wanted to get more comfortable with exceptions and handling:    Philosophically, exceptions are going to happen. So your code has to be prepared for them anyway.   Once you've done that work, you might as well use them to your benefit and you will wind up with more resilient code.   This is where having good testing in the caller(s) will help ensure robustness.